onnxruntime/core/platform/posix/env.cc

```diff
@@ -203,13 +213,28 @@ class PosixThread : public EnvThread {
 
  private:
   static void* ThreadMain(void* param) {
-    std::unique_ptr<Param> p((Param*)param);
+    std::unique_ptr<Param> p(static_cast<Param*>(param));
     ORT_TRY {
+#if !defined(__APPLE__) && !defined(__ANDROID__) && !defined(__wasm__) && !defined(_AIX)
+      if (p->affinity_mask != std::numeric_limits<size_t>::max()) {
+        cpu_set_t cpuset;
+        CPU_ZERO(&cpuset);
+        CPU_SET(p->affinity_mask, &cpuset);
+        // We have no way of reporting or logging affinity failure
+        // and we are not at liberty of terminating the thread
+        // because it would make things worse.
```

Comment:
What's the motive of moving in the affinity setting into thread_main? And yes I should log affinity related errors since it will incur notable perf impact afterwards.

Reply:
Let's think about how to report the error either from a constructor or from ThreadMain

Reply:
We can not throw from a constructor or call anything that might throw after the thread was created.
Also it is consistent with windows.